src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/add_panel/add_panel_flyout.tsx

```diff
@@ -36,14 +36,16 @@ import {
 import { IContainer } from '../../../../containers';
 import { EmbeddableFactoryNotFoundError } from '../../../../errors';
 import { GetEmbeddableFactories, GetEmbeddableFactory } from '../../../../types';
+import { SavedObjectFinderUi } from '../../../../../../../kibana_react/public';
 
 interface Props {
   onClose: () => void;
   container: IContainer;
   getFactory: GetEmbeddableFactory;
   getAllFactories: GetEmbeddableFactories;
   notifications: CoreSetup['notifications'];
-  SavedObjectFinder: React.ComponentType<any>;
+  savedObjects: CoreStart['savedObjects'];
+  uiSettings: CoreSetup['uiSettings'];
```

Comment:
for my own understanding: what's the difference between `CoreStart` and `CoreSetup`?

Reply:
That's not cleanly done by me, I reverted those changes. The difference between CoreSetup and CoreStart is that they are services available during different startup phases of plugins. For the particular case of  `uiSettings`, start and setup core should be identical so it wouldn't matter.

Reply:
You can find more information here: https://github.com/elastic/kibana/issues/52491

Reply:
Edit: Copied wrong link